src/ws/service/friendships_service.rs

```diff
@@ -21,141 +23,132 @@ use super::{
             }
-        }
+        });
+
+        log::info!("Returning generator for all friends for user {}", social_id);
+        Ok(generator)
     }
+
     #[tracing::instrument(name = "RPC SERVER > Get Request Events", skip(request, context))]
     async fn get_request_events(
         &self,
         request: Payload,
         context: Arc<SocialContext>,
-    ) -> RequestEvents {
+    ) -> Result<RequestEvents, FriendshipsServiceError> {
         // Get user id with the given Authentication Token.
         let user_id = get_user_id_from_request(
             &request,
             context.synapse.clone(),
             context.users_cache.clone(),
         )
-        .await;
-
-        match user_id {
-            Ok(user_id) => {
-                // Look for users requests
-                match context.db.db_repos.clone() {
-                    Some(repos) => {
-                        let requests = repos
-                            .friendship_history
-                            .get_user_pending_request_events(&user_id.social_id)
-                            .await;
-                        match requests {
-                            // TODO: Handle get user requests query response error. Ticket #81
-                            Err(err) => {
-                                log::debug!("Get Friends > Get User Requests > Error: {err}.");
-                                todo!()
-                            }
-                            Ok(requests) => {
-                                friendship_requests_as_request_events(requests, user_id.social_id)
-                            }
-                        }
+        .await?;
+
+        let social_id = user_id.social_id.clone();
+        log::info!("Getting requests events for user: {}", social_id);
+        // Look for users requests
+        match context.db.db_repos.clone() {
+            Some(repos) => {
+                let requests = repos
+                    .friendship_history
+                    .get_user_pending_request_events(&user_id.social_id)
+                    .await;
+                match requests {
+                    Ok(requests) => {
+                        log::info!("Returning requests events for user {}", social_id);
+                        Ok(friendship_requests_as_request_events(
+                            requests,
+                            user_id.social_id,
+                        ))
                     }
-                    // TODO: Handle repos None. Ticket #81
-                    None => {
-                        log::error!("Get Friends > Db Repositories > `repos` is None.");
-                        todo!()
+                    Err(err) => {
+                        log::error!(
+                            "Get request events > Get user pending request events > Error: {err}."
+                        );
+                        match err {
+                            Error::RowNotFound => Err(FriendshipsServiceError::NotFound),
```

Comment:
Actually, this is inaccurate because in `friendship_history.rs` we're doing ([link](https://github.com/decentraland/social-service/blob/8e2644c03d403ce02b54954e02639d910bf12d93/src/entities/friendship_history.rs#L159-L176)):
```
Err(Error::RowNotFound) => Ok(vec![]),
```
Is it ok? What do u think?

Reply:
mm, but why would a select all in the database return an error with row not found? 🤔

Reply:
The library doc says that this error occurs when "/// No rows returned by a query that expected to return at least one row". Here, we're fetching the **pending** request events for a given user, but the user may not have any incoming or outgoing pending requests.

Reply:
I think we should not consider it as an error since we do expect the query to return 0 rows sometimes. Therefore, I'm leaving this line of code as is: `Err(Error::RowNotFound) => Ok(vec![]),` and I've removed the error matching in `friendship_service.rs`